tests/nnstreamer_grpc/unittest_grpc.cc

```diff
@@ -0,0 +1,441 @@
+  g_free (name);
+
+  name = g_strconcat (NNS_GRPC_TENSOR_SINK_NAME, "_dummy", NULL);
+  factory = gst_element_factory_find (name);
+  EXPECT_TRUE (factory == NULL);
+  g_free (name);
+}
+
+/**
+ * @brief Test modes.
+ */
+typedef enum {
+  GRPC_MODE_BOTH = 0,
+  GRPC_MODE_SRC,
+  GRPC_MODE_SINK
+} TestMode;
+
+/**
+ * @brief Test options.
+ */
+typedef struct
+{
+  gboolean server;
+  gchar* host;
+  guint port;
+  guint fps;
+  tensor_type type;
+  TestMode mode;
+} TestOption;
+
+/**
+ * @brief Data structure for test.
+ */
+typedef struct
+{
+  GMainLoop *loop;  /**< main event loop */
+  GstElement *pipeline; /**< gst pipeline for test */
+} TestData;
+
+/**
+ * @brief Data for pipeline and test result.
+ */
+static TestData test_data;
+
+static gboolean DEFAULT_SERVER = TRUE;
+static char DEFAULT_HOST[] = "localhost";
+static guint DEFAULT_PORT = 55115;
+static guint DEFAULT_FPS = 10;
+static guint DEFAULT_OUT = 0;
+static tensor_type DEFAULT_TYPE = _NNS_UINT8;
+
+/**
+ * @brief Prepare test pipeline
+ */
+static gboolean
+_setup_pipeline (TestOption &option)
+{
+  gchar *str_pipeline;
+
+  switch (option.mode) {
+    case GRPC_MODE_SRC:
+      str_pipeline = g_strdup_printf (
+        "tensor_src_grpc name=src server=%s host=%s port=%u ! "
+        "other/tensor,dimension=(string)1:1:1:1,type=(string)%s,framerate=(fraction)%u/1 ! "
+        "fakesink",
+        option.server ? "TRUE" : "FALSE", option.host, option.port,
+        gst_tensor_get_type_string (option.type), option.fps);
+      break;
+    case GRPC_MODE_SINK:
+      str_pipeline = g_strdup_printf (
+        "videotestsrc ! video/x-raw,format=RGB,width=640,height=480,framerate=%u/1 !"
+        "tensor_converter ! tensor_sink_grpc name=sink server=%s host=%s port=%u",
+        option.fps, option.server ? "TRUE" : "FALSE", option.host, option.port);
+      break;
+    default:
+      return FALSE;
+  }
+
+  test_data.pipeline = gst_parse_launch (str_pipeline, NULL);
+  g_free (str_pipeline);
+
+  return test_data.pipeline != NULL ? TRUE : FALSE;
+}
+
+/**
+ * @brief set default option
+ */
+static void
+_set_default_option (TestOption &option)
+{
+  option.server = DEFAULT_SERVER;
+  option.host = DEFAULT_HOST;
+  option.port = DEFAULT_PORT;
+  option.fps = DEFAULT_FPS;
+  option.type = DEFAULT_TYPE;
+  option.mode = GRPC_MODE_BOTH;
+}
+
+/**
+ * @brief Test gRPC tensor_src get default property
+ */
+TEST (nnstreamer_grpc, src_get_property_default)
+{
+  TestOption option;
+  GstElement *src;
+  gboolean silent, server;
+  guint port, out;
+  gchar *host;
+
+  _set_default_option (option);
+  option.mode = GRPC_MODE_SRC;
+
+  ASSERT_TRUE (_setup_pipeline (option));
+
+  src = gst_bin_get_by_name (GST_BIN (test_data.pipeline), "src");
+  ASSERT_TRUE (src != NULL);
+
+  g_object_get (src, "silent", &silent, NULL);
+  EXPECT_TRUE (silent);
+
+  g_object_get (src, "server", &server, NULL);
+  EXPECT_EQ (server, DEFAULT_SERVER);
+
+  g_object_get (src, "host", &host, NULL);
```

Comment:
need to free host.
The rest of the tests need to be modified also.

Reply:
Got it. Thanks.